sstables/mx/writer.cc

```diff
@@ -840,6 +842,30 @@ void writer::add_pi_block() {
     _pi_write_m.offsets_serialized_size += offsets.size();
 
     _pi_write_m.blocks.emplace_back(std::move(block));
+
+    // auto-scale?
+    if (_pi_write_m.auto_scale_threshold // enabled
+            && _pi_write_m.blocks.size() >= _pi_write_m.auto_scale_threshold // crossed threshold
+            && !(_pi_write_m.blocks.size() % 2)) { // has even number of blocks
+        utils::chunked_vector<pi_block> new_blocks;
+        new_blocks.reserve(_pi_write_m.blocks.size() / 2);
+        blocks = measuring_output_stream();
+        offsets = measuring_output_stream();
+        auto it = _pi_write_m.blocks.cbegin();
+        while (it != _pi_write_m.blocks.cend()) {
+            block = *it++;
+            auto block2 = *it++;
+            block.last = std::move(block2.last);
+            block.width += block2.width;
+            block.open_marker = block2.open_marker;
```

Comment:
I think you should keep block.marker here.

Reply:
I had the impression that it contains the most recent range tombstone when the block ends, not when it starts.
I.e. we need to keep the second one, not the first.  (Maybe this is an existing issue and it should be the last one before the block begins as its name suggests).

Refs:
1) https://github.com/scylladb/scylla/blob/6a3494442e6e870620b53adf8194fa845d4a5acc/sstables/mx/writer.cc#L828-L833

2) https://github.com/scylladb/scylla/blob/6a3494442e6e870620b53adf8194fa845d4a5acc/sstables/mx/writer.cc#L1342-L1346

If we want to keep the last `_current_tombstone` before the block begins we'll to keep a copy of it in the `_pi_block_m`.

Reply:
No, the existing code and your change are correct.